lib/spack/docs/conf.py

```diff
@@ -95,10 +95,21 @@
     for cmd in sorted(command_names):
         index.write('   * :ref:`%s`\n' % cmd)
 
-
+#
 # Run sphinx-apidoc
-sphinx_apidoc(['-T', '-o', '.', '../spack'])
-os.remove('modules.rst')
+#
+# Remove any previous API docs
+# Read the Docs doesn't clean up after previous builds
+# Without this, the API Docs will never actually update
+#
+apidoc_args = [
+    'sphinx_apidoc',   # The first arugment is ignored
```

Comment:
This was an interesting discovery. I don't think it's actually a bug either. When you run `sphinx-apidoc` from the command line, the Python script runs:
```python
if __name__ == '__main__':
    main()
```
`main()` then processes `sys.argv[1:]` appropriately. But if you import `sphinx.apidoc` and try to run `main()` directly, `sys.argv[0]` is actually the first argument, not the name of the script. That explains why `-T` was previously being ignored and we were still getting a `modules.rst` file.